extensions/RabbitMQ/RabbitMqConfig.cs

```diff
@@ -37,4 +40,55 @@ public class RabbitMqConfig
     /// Default: 3600 second, 1 hour.
     /// </summary>
     public int MessageTTLSecs { get; set; } = 3600;
+
+    /// <summary>
+    /// How many times to dequeue a messages and process before moving it to a poison queue.
+    /// </summary>
+    public int MaxRetriesBeforePoisonQueue { get; set; } = 20;
+
+    /// <summary>
+    /// Suffix used for the poison queues.
+    /// </summary>
+    public string PoisonQueueSuffix { get; set; } = "-poison";
+
+    /// <summary>
+    /// Verify that the current state is valid.
+    /// </summary>
+    public void Validate()
+    {
+        if (this.MaxRetriesBeforePoisonQueue < 0)
+        {
+            throw new ConfigurationException($"RabbitQM: {nameof(this.MaxRetriesBeforePoisonQueue)} cannot be a negative number");
+        }
+
+        if (string.IsNullOrWhiteSpace(this.PoisonQueueSuffix))
+        {
+            throw new ConfigurationException($"RabbitMQ: {nameof(this.PoisonQueueSuffix)} is empty");
+        }
+
+        if (!IsValidQueueName(this.PoisonQueueSuffix))
+        {
+            throw new ConfigurationException($"RabbitMQ: {nameof(this.PoisonQueueSuffix)} is too long or contains invalid chars");
+        }
+    }
+
+    private static bool IsValidQueueName(string name)
```

Comment:
I would do these checks directly in `Validate()`, throwing two different and specific error messages -- that would make it easier to address the error, otherwise people will have to check this class to understand how